e2e/test/FileUploadE2ETests.cs

```diff
@@ -73,18 +73,16 @@ private async Task UploadFile(Client.TransportType transport, string filename, b
             using (ServiceClient serviceClient = ServiceClient.CreateFromConnectionString(Configuration.IoTHub.ConnectionString))
             {
                 FileNotificationReceiver<FileNotification> notificationReceiver = serviceClient.GetFileNotificationReceiver();
-
-                
                 using (FileStream fileStreamSource = new FileStream(filename, FileMode.Open, FileAccess.Read))
                 {
                     await deviceClient.UploadToBlobAsync(filename, fileStreamSource).ConfigureAwait(false);
                 }
 
                 FileNotification fileNotification = await VerifyFileNotification(notificationReceiver, testDevice.Id).ConfigureAwait(false);
 
+                // The following checks allow running these tests multiple times in parallel. 
```

Comment:
The problem with this change is if there is one file notification doesn't happen, we won't know which test failed.

Reply:
Good point - one would need to go through the large ETL file to figure that out.
1. Instead of just removing the assert, I'll replace it with a logging line.
2. I'll log the device name.

The debugging then would be a matter of looking at all parallel runs and creating a list of device names+filenames. Matching the ones that were supposed to be uploaded with the notifications will get you the device name and file name.